GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java

```diff
@@ -397,4 +401,28 @@ InputStream fetchImageFileFromS3(String filename) {
         }
     }
 
+    /**
+     * scans for and deletes Message entries that are more than one year old
+     */
+    private void purgeOldMessages() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.YEAR, ONE_YEAR_AGO);
+        log.info("Starting scan for Message entries older than: " + deadline.getTime());
+        MessageDao messageDao = new MessageDao();
+        List<Key> purgable = new ArrayList<>();
+        String cursor = "";
+        do { //Do this in batches - there might be half a million
+            List<Message> messageList = messageDao.listCreatedBefore(deadline.getTime(), cursor, 1000);
```

Comment:
Would it be possible to retrieve only message keys? Not sure if that is possible here

Reply:
That would be nice, like we do in some of the datascrips, but I searched the docs, and found no way to ask for just keys.

Reply:
would it be possible to use something like https://github.com/akvo/akvo-flow/blob/develop/GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java#L345

Reply:
Sure. I just hate to add one more flavour of datastore-specific code when we are heading to SQL. But since it won't be me who has to refactor it, I'll rewrite it a third time...